enhancements/aws-ipv6-dev-env.md

```diff
@@ -0,0 +1,158 @@
+---
+title: AWS-IPv6-Dev-Env
+authors:
+  - "@russellb"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-10-24
+last-updated: 2019-10-24
+status: implementable
+---
+
+# aws-ipv6-dev-env
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+While Kubernetes itself includes support for IPv6, the OKD distribution does
+not fully support it yet.  To enable developers across the many components that
+make up OKD, this proposal is to add support for enabling IPv6 for an AWS
+cluster.  This will also be used to create a CI job that can exercise IPv6
+support across OKD.
+
+## Motivation
+
+Supporting IPv6 across OKD will require collaboration and fixes across many
+components.  Many developers are already used to doing development against AWS
+clusters, so supporting AWS as an IPv6 development environment should make it
+easier for people to jump in to help debug and fix IPv6 issues.
+
+### Goals
+
+* Make creating an IPv6 development environment as easy as creating an AWS
+  cluster, by just setting one additional option to turn it on.
+* Support OKD in reaching a milestone of supporting a single stack IPv6 control
+  plane (IPv6 only).
+
+### Non-Goals
+
+* Reaching full support for IPv6 on AWS.  While the work will get us closer,
+  that is not the primary goal here.  The primary goal is to help make it
+  easier for all teams to exercise IPv6.
+
+## Proposal
+
+To enable IPv6 in your AWS environment, a developer would set the following
+environment variable before running the installer:
+
+```bash
+    export OPENSHIFT_AWS_USE_IPV6=”true”
+```
+
+This would be off by default.
+
+### AWS Network Environment
+
+AWS does not support a single-stack (IPv6 only) environment, but it does
+support a dual-stack (IPv4 and IPv6) environment, so that’s what is enabled
+here.  This is a summary of the changes to the network environment:
+
+* The VPC has IPv6 enabled and a `/56` IPv6 CIDR will be allocated by AWS.
```

Comment:
Maybe we can look at using bring-your-own-vpc for ipv6 networking on AWS. Because that already handles the case of user provided routes/natgw/igw etc..

Reply:
Wouldn't that just be *more* work?

AFAICT you can't get a 100% single-stack IPv6 environment on AWS. eg, "[We do not provide DNS hostnames for IPv6 addresses.](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html)"

But that's ok, this is helping us fix upstream for the dual-stack-but-preferring-IPv6 case, which we eventually need to support.